### ISSUE TITLE
Wasm module compression

### ISSUE THREAD

**Issue by Stebalien** [filecoin-project/fvm-specs#72]
Currently, our actors are pretty large (over 2MiB in some cases). Unfortunately, we _really_ want to keep a 2MiB upper limit on IPLD objects. If we don't fix this _now_, this will be a pain for us _later_ as we start getting stricter about maximum IPLD block sizes in transports, blockstores, etc.

One solution here would be to compress them. This can easily bring our larger wasm modules down to ~700KiB.

**Comment by dignifiedquire** [comment]
using `wasm-opt` I am getting these result

```sh
# original
2.0M - fil_actor_account.wasm
# -O3 (speed aggressive)
1.6M - fil_actor_account-3.wasm
# -Os (size)
1.6M - fil_actor_account-s.wasm
# -Oz (size aggressive)
1.6M - fil_actor_account-z.wasm
```

**Comment by Kubuxu** [comment]
Today for miner module:
serde_ipld_dagcbor::de::Deserializer<R>::recursion_checked monomorphization is 7% of the wasm module size: https://github.com/vmx/serde_ipld_dagcbor/blob/master/src/de.rs#L413-L424 this should be avoidable
serde_ipld_dagcbor::de::Deserializer<R>::parse_value is 10%.

**Comment by Stebalien** [comment]
@dignifiedquire:

1. Make sure to strip (wasm-strip).
2. Focus on the miner actor.

**Comment by Stebalien** [comment]
Yup. But with compression (zst), that goes down to 700kib, IIRC.

**Comment by raulk** [comment]
@Stebalien mind updating this with your latest discoveries re: compression?

**Comment by Stebalien** [comment]
For the miner actor (the largest):

1. After upgrading to the new IPLD library and stripping, we're now down to 1.3MiB by default.
2. If we optimize with `wasm-opt -O3 -Oz`, we can get down to 1015KiB.
3. Compressed with zstd, we can get down to 288KiB.

So we can deploy the miner actor in around 5 messages if we use compression, or 16 messages if we don't.
